preprocessor.py

```diff
@@ -75,32 +93,64 @@ def tag_usernames(text):
 
 
 def preprocess(
-    df, emojis, emoticons, urls, html_tags, acronyms, spelling, usernames,
+    in_filename,
+    out_filename,
+    emojis,
+    emoticons,
+    urls,
+    html_tags,
+    acronyms,
+    spelling,
+    usernames,
+    max_workers,
+    chunksize,
 ):
-    if emojis:
-        print("Removing emojis...")
-        df["text"] = df["text"].progress_apply(lambda text: remove_emojis(text))
-    if emoticons:
-        print("Removing emoticons...")
-        df["text"] = df["text"].progress_apply(lambda text: remove_emoticons(text))
+    df = pd.read_csv(in_filename)
+
+    if chunksize == -1:
+        if df.shape[0] < 1000:
+            chunksize = 1
+        else:
+            chunksize = min(df.shape[0] // max_workers, 1000)

```

Comment:
This sets the default value of chunksize to 1 for small list and 1000 for large lists of elements (it is faster). However, if the user specifies a chunksize different from the default value, the given value will be used regardless of efficiency.